test/Shape.test.ts

```diff
@@ -0,0 +1,171 @@
+  public getCurveEndIndices() { return []; }
+
+  // 添加 getter 方法来访问 protected 属性
+  public getFill() { return this.fill; }
+  public getStroke() { return this.stroke; }
+}
+
+describe("Shape", () => {
+  describe("restyle", () => {
+    test("updates material properties", () => {
+      const shape = new TestShape();
+      const newFill = new THREE.Color('#ff0000');
+      
+      shape.restyle({ 
+        fillColor: newFill,
+        fillOpacity: 0.5,
+        strokeWidth: 4
+      });
+
+      const fillMat = shape.getFill().material as THREE.MeshBasicMaterial;
+      expect(fillMat.color.getHexString()).toBe('ff0000');
+      expect(fillMat.opacity).toBe(0.5);
+
+      const strokeMat = shape.getStroke().material as any;
+      expect(strokeMat.width).toBe(4);
+    });
+
+    test("propagates to direct children when includeChildren is true", () => {
+      const parent = new TestShape();
+      const child = new TestShape();
+      const grandchild = new TestShape();
+      
+      parent.add(child);
+      child.add(grandchild);
+      
+      const newColor = new THREE.Color('blue');
+      parent.restyle({ fillColor: newColor }, { includeChildren: true });
+      
+      expect(parent.getFill().material.color.getHexString()).toBe('0000ff');
+      expect(child.getFill().material.color.getHexString()).toBe('0000ff');
+      expect(grandchild.getFill().material.color.getHexString()).not.toBe('0000ff');
+    });
+
+    test("ignores non-Shape children", () => {
+      const parent = new TestShape();
+      const light = new THREE.PointLight();
+      parent.add(light);
+      
+      parent.restyle({ fillColor: new THREE.Color('green') }, { includeChildren: true });
+      
+      expect(light.userData.fill).toBeUndefined();
+    });
+  });
+
+  test("material update", () => {
+    const shape = new TestShape([new Vector3()]);
+    const material = shape.getFill().material as MeshBasicMaterial;
+    material.color = new Color('red');
+    expect(material.color.getHexString()).toBe('ff0000');
+  });
+
+  test("restyle updates materials", () => {
+    const shape = new TestShape();
+    if (!shape.fill) {
+      throw new Error("Shape fill not initialized");
+    }
+
+    const newColor = new THREE.Color('red');
+    shape.restyle({ fillColor: newColor });
+    
+    expect(shape.fill.material.color.getHexString())
+      .toBe(newColor.getHexString());
+  });
+
+  test("includeChildren propagates to direct children", () => {
+    const parent = new TestShape();
+    const child = new TestShape();
+    const grandchild = new TestShape();
+    
+    if (!parent.fill || !child.fill || !grandchild.fill) {
+      throw new Error("Shapes missing required properties");
+    }
+
+    parent.add(child);
+    child.add(grandchild);
+    
+    const newColor = new THREE.Color('blue');
+    parent.restyle({ fillColor: newColor }, { includeChildren: true });
+    
+    expect(parent.getFill().material.color.getHexString()).toBe('0000ff');
```

Comment:
It'd be best to move these checks into a helper function to prevent repeating lines.